src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -618,6 +633,9 @@ protected bool EqualsHelper(CompilationOptions other)
                    this.ReportSuppressedDiagnostics == other.ReportSuppressedDiagnostics &&
                    string.Equals(this.ScriptClassName, other.ScriptClassName, StringComparison.Ordinal) &&
                    this.SpecificDiagnosticOptions.SequenceEqual(other.SpecificDiagnosticOptions, (left, right) => (left.Key == right.Key) && (left.Value == right.Value)) &&
+                   PerTreeDiagnosticOptions.SequenceEqual(other.PerTreeDiagnosticOptions, (left, right) =>
```

Comment:
>SequenceEqual [](start = 44, length = 13)

Is ```SequenceEqual``` sensitive to the order of elements? It doesn't feel right to require specific order for elements in a dictionary simply because this isn't something that can be reliable forced.